documentation/book/con-the-upgrade-process.adoc

```diff
@@ -0,0 +1,55 @@
+// This module is included in the following assemblies:
+//
+// assembly-upgrading-kafka-versions.adoc
+
+[id='con-the-upgrade-process-{context}']
+= The upgrade process
+
+Depending on the difference between the version being upgraded from and to, the upgrade process works in one of two ways:
+
+* A three phase update. This is typically used for upgrading between Kafka versions differing in the major or minor versions, for example 2.0.0 to 2.1.0.
+* A single phase update. This is typically used for upgrading between Kafka versions differing in the micro versions, for example 2.0.0 to 2.0.1.
+
+== Three phase update
+
+When the versions of Kafka use a different interbroker protocol version, or log message format version a three phase update is used.
+
+Phase 1::
+The existing cluster is reconfigured so that the brokers will communicate using the old version of the protocol and using the old message format.
+A rolling restart is necessary so that the brokers are using the new image.
```

Comment:
* We need a rolling update to ensure that all brokers are using the old interbroker protocol while rolling out the new image (otherwise the brokers using the new image will try to use the new interbroker protocol by default, which the old brokers won't understand).
* We need a 2nd rolling update to switch to the new interbroker protocol.
* The user then needs to update the client apps, 
* Then we need a final rolling update to switch to the new message version.

Reply:
Ok, that makes sense I guess.

Reply:
Related to the first bullet I added a comment in the needed user action. I guess that user needs to set the interbroker protocol version to the old one before bumping the new `version` otherwise the kafka brokers will restart talking the new interbroker protocol version.